File: api/Medico.Api/DB/Models/VitalSigns.cs
using System;

namespace Medico.Api.DB.Models
{
    public class VitalSigns : BaseEntity
    {
        public Guid? AdmissionId { get; set; }

        public Admission Admission { get; set; }

        public Guid PatientId { get; set; }

        public PatientDemographic Patient { get; set; }

        public double? Pulse { get; set; }

        public double? SystolicBloodPressure { get; set; }

        public double? DiastolicBloodPressure { get; set; }

        public string BloodPressurePosition { get; set; }

        public string BloodPressureLocation { get; set; }   

        public string OxygenSaturationAtRest { get; set; }

        public double? OxygenSaturationAtRestValue { get; set; }

        public int? RespirationRate { get; set; }

        public DateTime CreateDate { get; set; }
    }
}

File: client/src/dataModels/vitalSigns.ts
import { BaseEntityModel } from "./baseEntityModel";
import { DateConverter } from "../helpers/dateConverter";

export class VitalSigns extends BaseEntityModel {
    Pulse: number;
    BloodPressurePosition: string;
    BloodPressureLocation: string;
    SystolicBloodPressure: number;
    DiastolicBloodPressure: number;
    OxygenSaturationAtRest: string;
    RespirationRate: number;
    OxygenSaturationAtRestValue: number;
    PatientId: string;
    AdmissionId: string;
    CreateDate: any;

    constructor(id: string = "", isDelete: boolean = false,
        patientId: string = "", admissionId = "") {

        super(id, isDelete);

        this.PatientId = patientId;
        this.AdmissionId = admissionId;
        this.Pulse = null;
        this.OxygenSaturationAtRestValue = null;
        this.SystolicBloodPressure = null;
        this.OxygenSaturationAtRest = "";
        this.DiastolicBloodPressure = null;
        this.RespirationRate = null;
        this.BloodPressurePosition = "";
        this.BloodPressureLocation = "";
        this.CreateDate = new Date();
    }

    convertToEntityModel() {
        this.CreateDate = DateConverter
            .jsLocalDateToSqlServerUtc(this.CreateDate);
    }

    createFromEntityModel(entity: any) {
        const newVitalSigns = new VitalSigns(entity.Id, !!entity.isDelete, entity.PatientId, entity.AdmissionId);
        
        newVitalSigns.Pulse = entity.Pulse;
        newVitalSigns.SystolicBloodPressure = entity.SystolicBloodPressure;
        newVitalSigns.DiastolicBloodPressure = entity.DiastolicBloodPressure;
        newVitalSigns.BloodPressurePosition = entity.BloodPressurePosition;
        newVitalSigns.BloodPressureLocation = entity.BloodPressureLocation;
        newVitalSigns.OxygenSaturationAtRest = entity.OxygenSaturationAtRest;
        newVitalSigns.OxygenSaturationAtRestValue = entity.OxygenSaturationAtRestValue;
        newVitalSigns.RespirationRate = entity.RespirationRate;

        newVitalSigns.CreateDate = DateConverter
            .sqlServerUtcDateToLocalJsDate(entity.CreateDate);

        return newVitalSigns;
    }
}